### PR TITLE
Expose more fields from commandInfo to hooks

### PR DESCRIPTION
**Pull request by kamaradclimber**
Change-Id: I108554cb3cd236fec2edd97327c9f8305da83b47

### CODE REVIEW

**Review by pierrecdn — COMMENTED** [review]
I now understand why you were talking about executorInfo yesterday :)

FAR: 
I'm afraid you'll have to map all the mesos.proto into ruby hashes at the end since reflection in C++ is kind of hard AFAIK.

How does the Python/Java native(v0) bindings deal with that?
Maybe a starting point here? https://github.com/apache/mesos/blob/master/src/python/native_common/common.hpp

File: src/RubyHook.cpp [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 #include "RubyHook.hpp"
 #include <stdexcept>
+#include<string>
```
Comment (clems4ever) [inline src/RubyHook.cpp]: space

File: include/RubyEngine.hpp [hunk 2]
```diff
@@ -14,6 +14,7 @@ namespace criteo {
     namespace helpers{
       void hash_set(VALUE hash, const std::string& key, const std::string& value);
       void hash_set(VALUE hash, const std::string& key, VALUE value);
+      void ary_push(VALUE hash, const std::string& el);
```
Comment (clems4ever) [inline include/RubyEngine.hpp:17]: Why is Ruby the only one using this "ary" nomenclature for arrays?! ....... It's the first time I see it...
Just complaining, nothing to do about it in this review.

File: src/RubyHook.cpp [hunk 3]
```diff
@@ -104,11 +83,47 @@ VALUE wrapExecutorInfo(const mesos::ExecutorInfo& executorInfo)
...
   }
   hash_set(hash, "command", command);
 
   return hash;
 }
+
+// map parts of mesos::TaskInfo into a Ruby hash structure
+VALUE wrapTaskInfo(const mesos::TaskInfo& taskInfo)
+{
+  // create top level hash with required PB fields
```
Comment (clems4ever) [inline src/RubyHook.cpp:104]: what does PB mean?

**Review by clems4ever — CHANGES_REQUESTED** [review]
Overall, I'm fine with the changes. Just 2 small things.

**Comment by clems4ever** [comment]
@pierrecdn , Christophe decided to only provide what was needed from the TaskInfo to the module but I guess there is no good reason apart from we don't want to expose more than needed. Therefore we need to add any info we would require in our modules and that is not exposed yet. 

AFAIC, I'd be ok to use something like this https://github.com/ruby-protobuf/protobuf to convert the serialized task info into a ruby object and expose every attributes since we would have access to everything in a pure C++ module anyway.

**Comment by pierrecdn** [comment]
@clems4ever that was my thought, but since it's open-source I guess it could gain visibility at doing similar things, probably by using the ruby-protobuf you mentioned.
Definitely out of the scope of this review :smile:

**Comment by kamaradclimber** [comment]
Yes we will migrate to protobuf at some point

File: src/RubyHook.cpp (hunk 3 above)
Comment (kamaradclimber) [reply src/RubyHook.cpp:104]: protobuf

**Pull request merged**